ospfd/ospf_spf.c

```diff
@@ -1464,8 +1464,13 @@ static void ospf_spf_next(struct vertex *v, struct ospf_area *area,
 			if (ospf_nexthop_calculation(area, v, w, l, distance,
 						     lsa_pos))
 				vertex_pqueue_add(candidate, w);
-			else if (IS_DEBUG_OSPF_EVENT)
-				zlog_debug("Nexthop Calc failed");
+			else {
+				listnode_delete(area->spf_vertex_list, w);
+				ospf_vertex_free(w);
+				w_lsa->stat = LSA_SPF_NOT_EXPLORED;
```

Comment:
Isn't this statement redundant here? The value is already LSA_SPF_NOT_EXPLORED as per Line 1459

Reply:
w_lsa->stat is set to w in line 208, so need to clear it